java/semantickernel-core/src/main/java/com/microsoft/semantickernel/templateengine/DefaultPromptTemplateEngine.java

```diff
@@ -76,13 +78,15 @@ public Mono<String> renderAsync(List<Block> blocks, SKContext context) {
                             } else if (block instanceof CodeRendering) {
                                 return ((CodeBlock) block).renderCodeAsync(context);
                             } else {
-                                // const string error = "Unexpected block type, the block doesn't
-                                // have a rendering method";
-                                //    this._log.LogError(error);
-                                //    throw new
-                                // TemplateException(TemplateException.ErrorCodes.UnexpectedBlockType, error);
+                                LOGGER.error(
+                                        "Unexpected block type, the block doesn't have a rendering"
+                                                + " method");
+                                return Mono.error(
+                                        new TemplateException(
+                                                TemplateException.ErrorCodes.UnexpectedBlockType,
+                                                "Unexpected block type, the block doesn't have a"
+                                                        + " rendering method"));
```

Comment:
Create a variable to contain the message. The string concatenation creates extra strings and it made worse that we have the same string split up different.

Reply:
the splitting up is an artifact of the code formatter enforcing line lengths and splitting the string to maintain a line length. This causes no issues as the compiler eliminates the concatenation, so with or without the concatenation you will end up with the same bytecode. But can extract it to a variable if you feel makes more readable code, as I say performance wise extracted, with or without concatenation, it all gets compiled to the same bytecode.